dask/dataframe/io/parquet/core.py

```diff
@@ -0,0 +1,452 @@
+    compression="default",
+    write_index=True,
+    append=False,
+    partition_on=None,
+    storage_options=None,
+    compute=True,
+    **kwargs
+):
+    """Store Dask.dataframe to Parquet files
+
+    Notes
+    -----
+    Each partition will be written to a separate file.
+
+    Parameters
+    ----------
+    df : dask.dataframe.DataFrame
+    path : string
+        Destination directory for data.  Prepend with protocol like ``s3://``
+        or ``hdfs://`` for remote data.
+    engine : {'auto', 'fastparquet', 'pyarrow'}, default 'auto'
+        Parquet library to use. If only one library is installed, it will use
+        that one; if both, it will use 'fastparquet'.
+    compression : string or dict, optional
+        Either a string like ``"snappy"`` or a dictionary mapping column names
+        to compressors like ``{"name": "gzip", "values": "snappy"}``. The
+        default is ``"default"``, which uses the default compression for
+        whichever engine is selected.
+    write_index : boolean, optional
+        Whether or not to write the index. Defaults to True *if* divisions are
+        known.
+    append : bool, optional
+        If False (default), construct data-set from scratch. If True, add new
+        row-group(s) to an existing data-set. In the latter case, the data-set
+        must exist, and the schema must match the input data.
+    partition_on : list, optional
+        Construct directory-based partitioning by splitting on these fields'
+        values. Each dask partition will result in one or more datafiles,
+        there will be no global groupby.
+    storage_options : dict, optional
+        Key/value pairs to be passed on to the file-system backend, if any.
+    compute : bool, optional
+        If True (default) then the result is computed immediately. If False
+        then a ``dask.delayed`` object is returned for future computation.
+    **kwargs
+        Extra options to be passed on to the specific backend.
+
+    Examples
+    --------
+    >>> df = dd.read_csv(...)  # doctest: +SKIP
+    >>> to_parquet('/path/to/output/', df, compression='snappy')  # doctest: +SKIP
+
+    See Also
+    --------
+    read_parquet: Read parquet data to dask.dataframe
+    """
+    partition_on = partition_on or []
+    if isinstance(partition_on, string_types):
+        partition_on = [partition_on]
+
+    if set(partition_on) - set(df.columns):
+        raise ValueError("Partitioning on non-existent column")
+
+    if compression != "default":
+        kwargs["compression"] = compression
+    elif "snappy" in compress:
+        kwargs["compression"] = "snappy"
+
+    if isinstance(engine, str):
+        engine = get_engine(engine)
+
+    fs, fs_token, _ = get_fs_token_paths(
+        path, mode="wb", storage_options=storage_options
+    )
+    # Trim any protocol information from the path before forwarding
+    path = infer_storage_options(path)["path"]
+
+    if write_index:
+        df = df.reset_index()
```

Comment:
`drop=True`; but actually, the engine may want to write the resultant default index anyway (perhaps as metadata), so likely this kwarg needs to be passed on.

Reply:
I think we actually want `drop=False` here.  My recommendation here is that we always ask the parquet writer to only write the columns and not bother with the index at all (which raises tricky questions).  If the user wants the index then we promote it to a column.

This is definitely a deviation from prior behavior though.  Push back welcome.  I felt that it simplified things, but perhaps capturing pandas index metadata is more useful than it is tricky.

Reply:
Agree it's tricky, but we will have to deal with it anyway. I think the arrow writer will now always write an index metadata piece even for the default range.

In any case, I think drop=False here would surprise users, since they say "don't write index", yet there it is in data data.

Reply:
> In any case, I think drop=False here would surprise users, since they say "don't write index", yet there it is in data data.

Well, the default here is `write_index=True`, so in order for them to drop their index data they're going to have to explicitly say `write_index=False`, which I think takes care of the surprise factor.